plugins/index.ts

```diff
@@ -184,10 +184,8 @@ const deployOrUpgradeProxy = async <
         }) as InstanceOfContract<TContract>;
       }
     } catch (error) {
-      hre.trace(`Failed to upgrade ${contractName} with error:`, error);
-      contract = hre.deployments.get(
-        contractName
-      ) as InstanceOfContract<TContract>;
+      hre.log(`Failed to upgrade ${contractName} with error:`, error);
+      throw new Error(`Failed to upgrade ${contractName} with error: ${error}`);
```

Comment:
Throwing here is actually important because otherwise a failed upgrade/deployment will still lead to deployments being saved/written using the artifacts from the failed upgrade, thus subsequent attempts to upgrade will think that there has been no change and therefore no need to upgrade.  We shouldn't write deployment info if something went wrong during the deployment. @scottynomad  just flagging you here in case you are aware of a reason why it would be problematic to throw here instead of swallowing this error.